plugins/plugin-chart-echarts/src/Graph/transformProps.ts

```diff
@@ -0,0 +1,191 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  getMetricLabel,
+  DataRecord,
+} from '@superset-ui/core';
+import { EChartsOption, GraphSeriesOption } from 'echarts';
+import { GraphNodeItemOption } from 'echarts/types/src/chart/graph/GraphSeries';
+import { EchartsGraphFormData, DEFAULT_FORM_DATA as DEFAULT_GRAPH_FORM_DATA } from './types';
+import { GraphConstants, tooltipConfig, normalizationLimits } from './constants';
+import { EchartsProps } from '../types';
+import { getChartPadding, getLegendProps } from '../utils/series';
+
+/* eslint-disable no-param-reassign */
+function setLabelVisibility(nodes: GraphNodeItemOption[], showSymbolThreshold: number): void {
+  if (showSymbolThreshold > 0) {
+    nodes.forEach(function (node) {
+      node.label = {
+        show: node.value! > showSymbolThreshold,
+      };
+    });
+  }
+}
+
+/* eslint-disable no-param-reassign */
+function setNormalizedSymbolSize(nodes: GraphNodeItemOption[]): void {
+  let minValue: any = Number.MAX_VALUE;
+  let maxValue: any = Number.MIN_VALUE;
+  nodes.forEach(node => {
+    if (node.value! > maxValue) {
+      maxValue = node.value;
+    }
+    if (node.value! < minValue) {
+      minValue = node.value;
+    }
+  });
+  nodes.forEach(node => {
+    node.symbolSize =
+      // @ts-ignore: value is not null
+      (((node.value - minValue) / (maxValue - minValue)) * normalizationLimits.nodeSizeRightLimit ||
+        0) + normalizationLimits.nodeSizeLeftLimit;
+  });
+}
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queriesData } = chartProps;
+  const data: DataRecord[] = queriesData[0].data || [];
+
+  const {
+    name,
+    source,
+    target,
+    category,
+    colorScheme,
+    metric = '',
+    layout,
+    roam,
+    draggable,
+    selectedMode,
+    showSymbolThreshold,
+    edgeLength,
+    gravity,
+    repulsion,
+    friction,
+    legendMargin,
+    legendOrientation,
+    legendType,
+    showLegend,
+  }: EchartsGraphFormData = { ...DEFAULT_GRAPH_FORM_DATA, ...formData };
+
+  const metricLabel = getMetricLabel(metric);
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  const nodes: { [name: string]: number } = {};
+  const echartNodes: GraphNodeItemOption[] = [];
+  const echartLinks: { source: string; target: string }[] = [];
+  const echartCategories: string[] = [];
+  let index = 0;
+  let sourceIndex = 0;
+  let targetIndex = 0;
+
+  data.forEach(link => {
+    const nodeSource: any = link[source];
+    const nodeTarget: any = link[target];
+    const nodeCategory: string =
+      category && link[category] ? link[category]!.toString() : 'default';
+    const nodeValue: any = link[metricLabel];
```

Comment:
```suggestion
    const nodeValue = link[metricLabel] as number | null;
```

Reply:
set as GraphDataValue

Reply:
Maybe this is more accurate typing in terms what ECharts can accept:

```ts
    const nodeValue = link[metricLabel] as GraphNodeItemOption['value'];
```

But since node values are always numbers, this should work for us better:

```ts
  type EChartGraphNode = GraphNodeItemOption & {
    value: number;
  };

  const echartNodes: EChartGraphNode[] = [];
  // ...

  const nodeValue = link[metricLabel] as number;
```

Then you can remove all the non-null assertions (`!`).